File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxi=0;
        int l=0,r=height.length-1;

        while(l<r){
            maxi=Math.max(maxi,Math.min(height[r],height[l])*(r-l));
            if(height[l]<height[r]){
                l++;
            }else if(height[r]<height[l]){
                r--;
            }else{
                l++;
                r--;
            }
        }
        return maxi;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        maxi=-float('inf')
        l=0
        r=len(nums)-1

        while(l<r):
            maxi=max(maxi,(r-l)*min(nums[l],nums[r]))

            if nums[l]<nums[r]:
                l+=1
            elif nums[l]>nums[r]:
                r-=1
            else:
                r-=1
                l+=1
        return maxi